### ISSUE TITLE
New teams styles: pick languages before joining

### ISSUE THREAD

**Issue by bendk** [pculture/unisubs#2192]
If a user wants to join a team, they should have to pick which languages they speak first.

In the old code we used a modal dialog, we could re-implement that.

I think things would be simpler if we added an intermediate page instead of the dialog.

**Comment by PCF-Testing** [comment]
@bendk: @kslottow: We do have this page for new teams, but it needs v.1.9 re-styling.

**Comment by PCF-Testing** [comment]
Current layout:

![screenshot from 2018-09-28 07 36 10](https://user-images.githubusercontent.com/288455/46215112-66a81b00-c2f1-11e8-9eba-ad84ae3864a1.png)

**Comment by danantonioreyes** [comment]
@kslottow this seems good but there is just one slight quirk: it seems that a language-less member will get redirected to the language selection page when they try to access the dashboard or the videos page (they can access the activity and the members pages)

With the proposed changes, are we going to just redirect to the "language-less version of the dashboard" when the member attempts to go to the videos page?

**Comment by kslottow** [comment]
I think we should show the same pseudo-zero state on the available assignments page for collab teams and the simple teams video page, unless that's a lot more work.

**Comment by danantonioreyes** [comment]
Implemented in https://github.com/pculture/unisubs/tree/gh-enterprise-2192

**Comment by Rachel36** [comment]
Hi @danantonioreyes! I love the new look of this page. :+1:  Just a couple of notes from testing today: 

- [ ] Just to confirm, this issue does **not**  apply to the member profile language-selector, correct?  We are only looking at team pages here, correct? 

![image](https://user-images.githubusercontent.com/42118332/48438571-e0815000-e749-11e8-8c97-50fc275bd35a.png)

- [ ] If I select the same language multiple times, I get redirected to an error page:

![image](https://user-images.githubusercontent.com/42118332/48438709-2ccc9000-e74a-11e8-9433-c52661a6a09c.png)
 
![image](https://user-images.githubusercontent.com/42118332/48438723-335b0780-e74a-11e8-92cb-988059ff3018.png)

- [ ] A language-less user is currently able to access the activity and members pages.  Just want to confirm this is correct (looking at you and @kslottow's comments above)?

**Comment by kslottow** [comment]
> A language-less user is currently able to access the activity and members pages. Just want to confirm this is correct (looking at you and @kslottow's comments above)?

I was thinking they should see zero states on pages where they would need to have their languages selected to perform an action, such as taking an assignment. Those pages are:
- available assignments page for collab teams (videos tab -> assignments subtab)
- team videos page for simple teams (videos tab -> videos subtab)

For the other pages, like activity and members pages, the user doesn't need to pick their language to use the page properly. Does that seem right to you, @Rachel36 @danantonioreyes ?

**Comment by kslottow** [comment]
@danantonioreyes did you decide against this idea (from description), in favor of putting a form on the dashboard if the user doesn't have a language selected in their profile? I'm not totally attached to it, but curious about your reasoning if you decided the form was better:
- if a user doesn't have a language selected in their profile, we replace the "Finding Work" section (My assignments and Available Assignments for collab teams, Videos for simple teams) with the linked text, "After you select the languages you speak, you'll see suggested videos to subtitle here."
- "select the languages you speak" can be linked, the rest can be helptext.
- the other sections on the dashboard should remain visible

![image](https://user-images.githubusercontent.com/42118332/48438709-2ccc9000-e74a-11e8-9433-c52661a6a09c.png)

**Comment by danantonioreyes** [comment]
@kslottow this is what the collab/simple team dashboards look like when the team member has no profile languages yet:

![image](https://user-images.githubusercontent.com/1912705/48482911-eb70d900-e84c-11e8-8b6c-70b3170e690a.png)

The screenshot above is the page that is linked in the zero-state

**Comment by danantonioreyes** [comment]
@Rachel36 

> Just to confirm, this issue does not apply to the member profile language-selector, correct? We are only looking at team pages here, correct?

Yes this does not apply to the language selectors in the user profile page

> A language-less user is currently able to access the activity and members pages. Just want to confirm this is correct (looking at you and @kslottow's comments above)?

> For the other pages, like activity and members pages, the user doesn't need to pick their language to use the page properly. Does that seem right to you, @Rachel36 @danantonioreyes ?

Yes it should be fine to see the activity and member pages even if the user does not have profile languages yet

**Comment by Rachel36** [comment]
Thanks for feedback @danantonioreyes and @kslottow! This one is looking good - pending update on what might be causing the error when I select the same language twice?

**Comment by kslottow** [comment]
@danantonioreyes I love the way the dashboard zero state looks!

I noticed I was really unclear about where this zero state should link TO in the description. I need to get better about this. While the new language selection page is beautiful (I would like to use it to replace the current selector on the Application Page), it seems to unnecessarily complicate the interaction.

Here's what I was thinking:
1. The user goes to their dashboard or available assignments page and sees they need to set their languages first, so they click a link 
2. They arrive on their user profile, where the language selection already exists. They then get used to the idea that this is where language selection always happens, rather than being able to select their languages somewhere else in the platform.

So I think we should link directly to the (old style) user profile from the text "select the languages you speak" in the zero state.

**Comment by danantonioreyes** [comment]
@kslottow linking to the user profile page sounds good, I haven't thought of that! One difference though would be that I'm not sure if I can redirect back to the team dashboard once the user sets his/her languages in the profile page-- I will have to check on this.

@Rachel36 since I'm removing the page where "select the languages you speak" links to, there won't be a need to address the double language error. I will double check in the old style language selector if this bug still exists

**Comment by danantonioreyes** [comment]
@Rachel36 this should be good for testing now. Specifying the same two languages does not seem to be a problem in the edit profile page.

Also added the zero-state message to the following:
For Simple teams, the Videos page should also show the same zero state
For Collab teams, the Assignments page (Assignments tab only, not the Videos tab) should also show the same zero state

**Comment by danantonioreyes** [comment]
Good catch Rachel! Implemented the check for both collab and simple teams

**Comment by Rachel36** [comment]
Changes look good @danantonioreyes!  2192 has been verified.

**Comment by danantonioreyes** [comment]
@Rachel36 @PCF-Testing to whomever will be testing this in dev, please also test for cases where the user is not actually logged in -- this is relevant because teams can set their team visibility and video visibility to public

**Comment by PCF-Testing** [comment]
@danantonioreyes: on the subtitle language page, the redirect is to /teams/ page instead of the current team - https://dev.amara.org/en/profiles/edit/?next=/teams/. See, for example, on page https://dev.amara.org/en/videos/wr2HgCDmjA0R/fr/1093795/?team=m-collab-test as user test2192.

**Comment by danantonioreyes** [comment]
Fixed in commit https://github.com/pculture/unisubs/commit/54ad63f591534a2ed324996b29e72d3cbb07988a to dev

**Comment by PCF-Testing** [comment]
Confirming the fix.

**Comment by PCF-Testing** [comment]
The latest fix looks fine to me.
